### DISCUSSION TITLE
API for digital photo frames

### DISCUSSION THREAD

**Discussion by lanhar1** [ente-io/ente#644] [Enhancements]
I'm looking to get away from using our Google Home device for photo albums and would LOVE to be able to utilize my Ente albums for digital photo frames.

Don't know if something like that would require just a simple API or what, but figured I'd at least throw the idea out.

**Comment by vishnukvmd** [comment]
Hey @lanhar1, since Ente is e2ee, we don't have an API to pull photos directly from.

Does your photo frame accept a webpage / embeddable iframe?

If you could share more details about the API it can work with, we'd be happy to figure out what's possible.

**Reply by bradall** [reply]
Hi,

Would it be possible to add a slideshow view to the embedable option?
Embedding the album is good, but it doesnt really work for digital photo frame applications when you have to manually click in to full screen and swipe between photos. 
Thank you for the great work, ente is brilliant.

**Comment by lanhar1** [comment]
I'm not entirely sure, as I haven't actually built one yet or looked into any of the projects out there in great detail, just had the idea that it'd be cool to utilize my Ente account. I will have to do some research and see if and how they accept image sources.

**Comment by paulxb** [comment]
I'm looking at doing something similar. I am going to use [MagicMirror](https://magicmirror.builders/) on a RaspberryPi and probably use the CLI to sync the photos (though I haven't had a change to try it out yet!)

**Comment by creedda** [comment]
This is something the ente team should consider for long term growth of the platform.

It is not feasible to scale this out for each individual photo frame, but if the team provided easy to implement API calls for any open source or closed source photo frame to implement then it is more likely to be picked up by photo frame makers.

This helps with increasing market share and penetration with the general public.

Something like this maybe? https://github.com/googlesamples/google-photos/tree/main/REST/PhotoFrame

Basically we need an easy API call specifically for us to request the contents of an a specific album specifically for Photo Frames (for closed source) and PR (for open source) into Frame Makers list of supported photo platforms.


Selfishly and personally, the elder folks in my family use https://www.pix-star.com/ as their digital photo frames of choice. Would love to see ente albums added as an option.

In the meantime for those looking for an answer, I have Ente syncing with my NAS using the Ente CLI, and my NAS syncing to Dropbox which the photo frame pulls from.

**Comment by privateuser917** [comment]
Any updates on an Ente compatible photo viewer yet?

**Comment by kissu** [comment]
I think that having something like this: https://github.com/immichFrame/ImmichFrame
Would be quite amazing for Ente. It's actually one of the reasons why I'm still hesitating between both projects haha. 🤗

I guess that a self-hosted version + some coding skills could allow us to embed it somewhere like a Kodi or alike?
Would be amazing to have something on a big screen like the Apple TV screensaver. 🤩

Related discussion: https://github.com/ente-io/ente/discussions/8536

**Comment by BarheimCoder** [comment]
Found this while trying to find a solution. I guess the best way would be a view toggle on the app, then anything that can run Ente would be able to become a "live frame". The devices sold for this specific purpose are pretty much just tablets anyway.

There's definitely a market and demand from Ente users (including myself). I'd rather avoid relying on yet another app to get something like this setup. From a quick look:

- Amazon Firestick has integration with Amazon Photos to display photos in a similar way when a TV is idle
- Skylight, Frameo, DragonTouch are just some of the ones I could find online, each has its own app required to send photos to the frame. There are probably hundreds of these type forcing an app for any functionality.
- ImmicFrame as mentioned by kissu which is the closest competitor in the privacy-focused space.

**Reply by kissu** [reply]
Yeah I think that most people that are interested by Immich/Ente would also like to self-host their thing and not rely on a 3rd party mobile app or some server somewhere. 🤷🏻‍♀️
Not sure if there are technical limitations to the Ente implementation or something.

They do have a Discord channel: https://ente.io/help/#contributing
Or maybe asking during some livestream/YT video might maybe help? Not sure how to bump this one further otherwise.

**Reply by BarheimCoder** [reply]
Discord is for feedback on current features. I was advised to post here for suggestions like this. But good shout on asking during livestream / YT.

Getting visibility to this post to bump / upvote should also help gain traction and let them gauge better how it stacks against other features to explore.

**Comment by dawiddyd** [comment]
Amazing idea!

**Comment by aChrisYouKnow** [comment]
Came here looking for the same. Would love to have a frame interface which could just take in a list of ente public links and cycle through that mix of photos. Being able to just adjust the list of ente album links as needed.
